model/transfer.go

```diff
@@ -0,0 +1,127 @@
+package model
+
+import (
+	"database/sql"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/jinzhu/gorm/dialects/postgres"
+)
+
+type (
+	// TransferDestination schema for transfer destination
+	TransferDestination struct {
+		Type            string          `json:"type"`
+		BankDetails     BankDetails     `json:"bank_details"`
+		PersonalDetails PersonalDetails `json:"personal_details"`
+	}
+
+	// InitiateTransferRequest schema for initiate transfer request
+	InitiateTransferRequest struct {
+		CustomerID  string              `json:"customer_id"`
+		Amount      float64             `json:"amount"`
+		Currency    string              `json:"currency"`
+		Destination TransferDestination `json:"destination"`
+		Note        string              `json:"note,omitempty"`
+		Reason      string              `json:"reason"`
+		Reference   string              `json:"reference"`
+	}
+
+	// InitiateTerminalTransferRequest schema for initiate terminal transfer request
+	InitiateTerminalTransferRequest struct {
+		Amount              float64              `json:"amount"`
+		SourceCurrency      string               `json:"source_currency"`
+		DestinationCurrency string               `json:"destination_currency"`
+		UseBalance          string               `json:"use_balance"`
+		BeneficiaryID       *string              `json:"beneficiary_id,omitempty"`
+		Destination         *TransferDestination `json:"destination,omitempty"`
+		Note                *string              `json:"note,omitempty"`
+		Reason              string               `json:"reason"`
+	}
+
+	// TransferResponse schema for transfer response
+	TransferResponse struct {
+		ID uuid.UUID `json:"id"`
+		InitiateTransferRequest
```

Comment:
you can add a json tag

Reply:
It's returned like that from external api